quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/TruststoreDistTest.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.it.cli.dist;
+
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+import org.keycloak.it.junit5.extension.DistributionTest;
+import org.keycloak.it.junit5.extension.RawDistOnly;
+import org.keycloak.it.utils.KeycloakDistribution;
+import org.keycloak.it.utils.RawKeycloakDistribution;
+
+import java.nio.file.Path;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static io.restassured.RestAssured.given;
+
+@DistributionTest(keepAlive = true)
+@RawDistOnly(reason = "Containers are immutable")
+public class TruststoreDistTest {
+
+    @Test
+    void testMutualAuthWithGlobalTruststore(KeycloakDistribution dist) {
+        String[] truststoreNames = new String[] { "keycloak-truststore.p12", "self-signed.pem" };
+        Stream.of(truststoreNames).forEach(truststoreName -> {
+            dist.copyOrReplaceFileFromClasspath("/" + truststoreName, Path.of("conf", truststoreName));
+        });
+
+        RawKeycloakDistribution rawDist = dist.unwrap(RawKeycloakDistribution.class);
+        String paths = Stream.of(truststoreNames).map(truststoreName -> rawDist.getDistPath().resolve("conf")
+                .resolve(truststoreName).toAbsolutePath().toString()).collect(Collectors.joining(","));
+        dist.copyOrReplaceFileFromClasspath("/self-signed.p12", Path.of("conf", "self-signed.p12"));
+        Path keyStore = rawDist.getDistPath().resolve("conf").resolve("self-signed.p12").toAbsolutePath();
+
+        rawDist.run("--verbose", "start", "--http-enabled=true", "--hostname=mykeycloak.org",
+                "--truststore-paths=" + paths, "--https-client-auth=required", "--https-key-store-file=" + keyStore);
+
+        given().trustStore(TruststoreDistTest.class.getResource("/self-signed.p12").getPath(), "password")
+                .keyStore(TruststoreDistTest.class.getResource("/self-signed.p12").getPath(), "password")
+                .get("https://mykeycloak.org:8443").then().body(Matchers.containsString("https://mykeycloak.org"));
+
+        rawDist.stop();
```

Comment:
Just a nitpick. The dist is stopped [automatically](https://github.com/keycloak/keycloak/blob/83676bf927bf6430cbb6cff34a6f354cddd3a34d/quarkus/tests/junit5/src/main/java/org/keycloak/it/junit5/extension/CLITestExtension.java#L170).

Reply:
Ah, I picked that up from one of the other tests so there could be more cleanup - I'll remove from here at least.

Reply:
I see that I was referencing tests that started several servers in a loop - thus the need for the manual stop.